include/deal.II/lac/sparse_matrix.h

```diff
@@ -2376,6 +2383,42 @@ SparseMatrix<number>::end (const size_type r)
   return iterator(this, cols->rowstart[r+1]);
 }
 
+template<>
+template<class StreamType>
+void SparseMatrix<std::complex<double>>::print_matrix_market(StreamType &out,
```

Comment:
You need to replace `>>` by `> >` (not the space) here and in the next declaration of a specialization for C++98 compilers.